### DISCUSSION TITLE
Is it possible to use `transformer-variant-group` as plugin in Tailwind CSS?

### DISCUSSION THREAD

**Discussion by cawa-93** [unocss/unocss#2545] [Ideas]
I am forced to use tailwind, since it is "mainstream" and already added to the project. In addition, it has better support from editors. However, I really miss some features in it such as variant grouping.

Is it possible to somehow use `transformer-variant-group` as plugin in Tailwind CSS?

**Comment by zyyv** [comment]
In fact, `transformer-variant-group` exists only as a transformer, which runs in the vite plugin. If your project supports `vite`, you can give it a try.
You can also check the source code of `transformer-variant-group` and `@unocss/vite`, extract the parts that are useful to you, and then write a plugin to use.

**Reply by cawa-93** [reply]
It's kinda work
```js
// vite.config.js
import { expandVariantGroup } from '@unocss/core'

export default {
	plugins: [{
	    name: '@unocss/transformer-variant-group',
	    enforce: 'pre',
	    transform(s) {
	      return expandVariantGroup(s)
	    },
	}]
}
